packages/cli/src/error_handler.ts

```diff
@@ -0,0 +1,81 @@
+import { COLOR, Printer } from '@aws-amplify/cli-core';
+import { InvalidCredentialError } from './error/credential_error.js';
+import { EOL } from 'os';
+import { Argv } from 'yargs';
+
+/**
+ * Attaches process listeners to handle unhandled exceptions and rejections
+ */
+export const attachUnhandledExceptionListeners = (): void => {
+  process.on('unhandledRejection', (reason) => {
+    process.exitCode = 1;
+    if (reason instanceof Error) {
+      handleError(reason);
+    } else if (typeof reason === 'string') {
+      handleError(new Error(reason));
+    } else {
+      // This should never happen unless something truly strange has happened
+      Printer.print(`Cannot handle rejection of type [${typeof reason}]`);
```

Comment:
would `Promise.reject({})` land here?

Reply:
Good call, let me try that out

Reply:
Looks like that does land here. I'll run all cases through `handleError` and we can build up our library of how to handle stuff like that over time as needed.

Reply:
Perhaps we can just change error message to `Unexpected rejection of type ...` .

Reply:
or `Unexpected type of unexpected rejection ` :-)

Reply:
Updated to
```ts
handleError(
        new Error(`Unhandled rejection of type [${typeof reason}]`, {
          cause: reason,
        })
      );
```